ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/move/move_op.cpp

```diff
@@ -7,12 +7,19 @@
 namespace move_op_utils {
 using namespace tt::tt_metal;
 
-bool can_deallocate(const Tensor &input_tensor) {
+bool can_deallocate(const Tensor &input_tensor, bool from_multi_device) {
```

Comment:
@tt-aho could you give an explanation of the changes in this function?

Reply:
Our move op checks the ref count of a tensor to see if it can move or not. It expects the input to be the only reference.

The `bool from_multi_device` was added because with a multidevice tensor and launch_op, then there will be 2 references, one for the multidevice tensor, and one for the single device tensor inside launch_op. Otherwise to check for only one reference it would only work if the function is called outside launch_op.

(I will update `return storage.buffer.use_count() == 1 + (uint32_t)from_multi_device;` to `return storage.buffer.use_count() == (from_multi_device ? 2 : 1);`)

Reply:
This is a little bit sketchy to me because we're implicitly tying the use-count with where this function is being called. Is this the only place where it's currently being invoked?
https://github.com/tenstorrent/tt-metal/pull/10620/files#diff-1528f05378bae608f410b55bbd80f76c2abd5576d4bb7113caf0a47ae9498d51R123

Reply:
@tt-asaigal for any comments

Reply:
it is only used by `move` and `move_sharded` as a helper fn. I only updated `move_sharded` because `move` has not been updated to the `launch_op` api, otherwise it would also use this. Not expected to be used outside of this I think